### PR TITLE
Match commitment to specification

### PR DESCRIPTION
**Pull request by AdityaSripal**
## Description

closes: #7371 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/v2/types/commitment.go [hunk 1]
```diff
@@ -12,17 +12,23 @@ import (
 // NOTE: A fixed length preimage is ESSENTIAL to prevent relayers from being able
 // to malleate the packet fields and create a commitment hash that matches the original packet.
 func CommitPacket(packet Packet) []byte {
-	buf := sdk.Uint64ToBigEndian(packet.GetTimeoutTimestamp())
+	var buf []byte
+	timeoutBytes := sdk.Uint64ToBigEndian(packet.GetTimeoutTimestamp())
+	timeoutHash := sha256.Sum256(timeoutBytes)
+	buf = append(buf, timeoutHash[:]...)
 
 	destIDHash := sha256.Sum256([]byte(packet.DestinationChannel))
```
Comment (DimitrisJim) [inline modules/core/04-channel/v2/types/commitment.go:20]: Looking at https://github.com/cosmos/ibc/pull/1152/files as reference, is order in which we add to the buf array correct? We do: `[0x2, timeoutHash, destIDHash, appHash]` while in spec `[0x2, destIDHash, timeoutHash, appHash]`

**Review by DimitrisJim — COMMENTED** [review]
some comments, maybe I'm looking at wrong spec PR :sweat_smile:

**Comment by AdityaSripal** [comment]
Ahh yes, you're correct. Fixed now

**Review by DimitrisJim — APPROVED** [review]
lgtm! pushed a cheeky commit to update documentation

**Pull request merged**